### DISCUSSION TITLE
Given, When, And, Then aliases for Spec

### DISCUSSION THREAD

**Discussion by CaringDev** [bbvch/LambdaTale#37] [Polls]
In #30 @iamkoch brings up the idea of aliasing the new Spec to Given, When, Then, And, which would allow writing tests such as:
```csharp
Given("old state", () => ...);
And("some precondition", () => ...);
When("stimulus", () => ...);
Then("new state", () => ....);
```

**Comment by iamkoch** [comment]
My tests often use BDD-style language, so I end up with 

```
"Given some context".x(() =>
"And some more".x() =>
```

This would a little syntactic sugar for me and others like me, plus allow for extensions in the future where people can now explicitly hook into the before, action and after sections of a scenario.

**Comment by CaringDev** [comment]
Solved as https://github.com/bbvch/LambdaTale?tab=readme-ov-file#static-spec-method in https://github.com/bbvch/LambdaTale/pull/35

**Reply by iamkoch** [reply]
Awesome - thanks for this!
